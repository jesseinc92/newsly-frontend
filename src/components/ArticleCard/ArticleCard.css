.ArticleCard-link {
  text-decoration: none;
  color: black;
}

.ArticleCard {
  width: 75%;
  margin: 1rem auto;
  padding: 1rem;
  border: 1px solid black;
  border-radius: 1rem;
}

.ArticleCard-title {
  text-align: left;
  font-family: 'Old Standard TT', serif;
}

.ArticleCard-cat-badge {
  width: fit-content;
  margin: auto;
  padding: 0 1.35rem;
  font-size: 1.1rem;
  border: 2px solid;
  border-radius: 1rem;
}

/* Tablet sizing */
@media screen and (min-width: 768px) {
  
  .ArticleCard-cat-badge {
    font-size: 1.3rem;
    border-radius: 1.2rem;
  }

}

/* Laptop/Desktop sizing */
@media screen and (min-width: 1024px) {

  .ArticleCard {
    width: 100%;
    height: 100%;
    margin: 0;
  }

}