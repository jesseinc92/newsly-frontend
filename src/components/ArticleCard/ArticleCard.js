import React from 'react';
import { Link } from 'react-router-dom';
import './ArticleCard.css';

const ArticleCard = ({ article }) => {

  const escapedId = article.id.replaceAll('/', '%2F');

  return (
    <Link className='ArticleCard-link' to={`/article/${escapedId}`}>
      <div className='ArticleCard'>
        <p>{ article.webTitle }</p>
        <p>{ article.sectionName }</p>
      </div>
    </Link>
  )
}

export default ArticleCard;