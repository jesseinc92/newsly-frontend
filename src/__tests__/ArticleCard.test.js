import React from 'react';
import { render } from '@testing-library/react';
import ArticleCard from '../components/ArticleCard/ArticleCard';
import { MemoryRouter } from 'react-router-dom';

const article = {
    id: 'id',
    webTitle: 'title',
    sectionName: 'section'
}

it('renders the component', function() {
  

  render(<MemoryRouter>
          <ArticleCard article={article} />
        </MemoryRouter>);
});

it('matches the snapshot', function() {
  const { asFragment } = render(<MemoryRouter>
                                  <ArticleCard article={article} />
                                </MemoryRouter>);
  expect(asFragment()).toMatchSnapshot();
})