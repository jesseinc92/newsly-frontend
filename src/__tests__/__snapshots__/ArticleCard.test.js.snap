// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <a
    class="ArticleCard-link"
    href="/article/id"
  >
    <div
      class="ArticleCard"
    >
      <p
        class="ArticleCard-title"
      >
        <strong>
          title
        </strong>
      </p>
      <p
        class="ArticleCard-cat-badge undefined"
      >
        section
      </p>
    </div>
  </a>
</DocumentFragment>
`;
